### PR TITLE
feature/PLUGIN-353-Schema-autodetection

### PR DESCRIPTION
**Pull request by ailegion**
added: Schema auto-detection support for file source plugins

JIRA: https://issues.cask.co/browse/PLUGIN-353

Related to PR: https://github.com/cdapio/hydrator-plugins/pull/1203

### CODE REVIEW

File: core-plugins/docs/File-batchsource.md [hunk 1]
```diff
@@ -20,6 +20,11 @@ If the format is 'blob', every input file will be read into a separate record.
 The 'blob' format also requires a schema that contains a field named 'body' of type 'bytes'.
 If the format is 'text', the schema must contain a field named 'body' of type 'string'.
 
+**Get Schema:** Auto-detects schema from file. Supported formats are: avro, parquet, csv, delimited,tsv.
```
Comment (CuriousVini) [inline core-plugins/docs/File-batchsource.md]: `Supported formats are: avro, parquet, csv, delimited,tsv.` - It also supports blob and text right?

**Review by CuriousVini — COMMENTED** [review]
One comment on doc. Also why was the file test failing?

File: core-plugins/docs/File-batchsource.md (hunk 1 above)
Comment (ailegion) [reply core-plugins/docs/File-batchsource.md]: fixed

**Comment by ailegion** [comment]
> One comment on doc. Also why was the file test failing?

@CuriousVini the test was failing because it was designed to test null schema scenario and with this PR when schema is null we try to generate it from file. However the test was designed to provide only the file name for validation but not actually generate the file before the final assert. This caused 'File not found exception' during schema detection. We moved the file generation before the validation in order to make file available for schema detection.

**Review by CuriousVini — APPROVED** [review]

**Comment by CuriousVini** [comment]
Please squash the commits

**Comment by ailegion** [comment]
> Please squash the commits

squashed

**Pull request merged**